tests/fakeprotocol_iot.py

```diff
@@ -426,9 +433,9 @@ async def send(self, request, port=9999):
         # collect child ids from context
         try:
             child_ids = request["context"]["child_ids"]
-            request.pop("context", None)
         except KeyError:
             child_ids = []
+        request.pop("context", None)
```

Comment:
The `defaultDict` used by dump_devinfo for the final request caused the `context` to be populated by this check, so the `pop` is moved outside the try block.

Reply:
Do we really need to pop it out?

Reply:
Yes, it needs to be popped out as it's not available in the fixture and that's the way the fakeprotocol_iot.py is implemented. i.e. it was already being popped out but the `defaultDict` in `dump_devinfo` causes it to be created erroneously during the check.

Reply:
Actually this has been reverted due to the simplification